imports/services/isEditedService.js

```diff
@@ -0,0 +1,279 @@
+                    detail.isEditedDate = null;
+                }
+                else {
+                    if (detail.isEditedBy === Meteor.userId()) {
+                        detail.isEditedBy = null;
+                        detail.isEditedDate = null;
+                    }
+                }
+            }
+        }
+    }
+    minute.save();
+}
+
+function setIsEditedTop(minutesId, topicId) {
+    let topic = new Topic(minutesId, topicId);
+
+    topic._topicDoc.isEditedBy = Meteor.userId();
+    topic._topicDoc.isEditedDate = new Date();
+
+    topic.save();
+}
+
+function removeIsEditedTop(minutesId, topicId, any) {
+    let unset = false;
+    let topic = new Topic(minutesId, topicId);
+
+    if (any === true) {
+        unset = true;
+    }
+    else {
+        if (topic._topicDoc.isEditedBy === Meteor.userId()) {
+            unset = true;
+        }
+    }
+
+    if (unset === true) {
+        topic._topicDoc.isEditedBy = null;
+        topic._topicDoc.isEditedDate = null;
+        topic.save();
+    }
+
+}
+
+function setIsEditedII(minutesId, topicId, infoItemId) {
+    let topic = new Topic(minutesId, topicId);
+    let infoItem = topic.findInfoItem(infoItemId);
+
+    infoItem._infoItemDoc.isEditedBy = Meteor.userId();
+    infoItem._infoItemDoc.isEditedDate = new Date();
+
+    infoItem.save();
+}
+
+function removeIsEditedII(minutesId, topicId, infoItemId, any) {
+    let unset = false;
+    let topic = new Topic(minutesId, topicId);
+    let infoItem = topic.findInfoItem(infoItemId);
+
+    if (any === true) {
+        unset = true;
+    }
+    else {
+        if (infoItem._infoItemDoc.isEditedBy === Meteor.userId()) {
+            unset = true;
+        }
+    }
+
+    if (unset === true) {
+        infoItem._infoItemDoc.isEditedBy = null;
+        infoItem._infoItemDoc.isEditedDate = null;
+        infoItem.save();
+    }
+
+
+}
+
+function setIsEditedDet(minutesId, topicId, infoItemId, detailIdx) {
+    let topic = new Topic(minutesId, topicId);
+    let infoItem = topic.findInfoItem(infoItemId);
+
+    infoItem._infoItemDoc.details[detailIdx].isEditedBy = Meteor.userId();
+    infoItem._infoItemDoc.details[detailIdx].isEditedDate = new Date();
+
+    infoItem.save();
+}
+
+function removeIsEditedDet(minutesId, topicId, infoItemId, detailIdx, any) {
+    let unset = false;
+    let topic = new Topic(minutesId, topicId);
+    let infoItem = topic.findInfoItem(infoItemId);
+
+    if (any === true) {
+        unset = true;
+    }
+    else {
+        if (infoItem._infoItemDoc.details[detailIdx].isEditedBy === Meteor.userId()) {
+            unset = true;
+        }
+    }
+
+    if(unset === true) {
+        infoItem._infoItemDoc.details[detailIdx].isEditedBy = null;
+        infoItem._infoItemDoc.details[detailIdx].isEditedDate = null;
+        infoItem.save();
+    }
+
+
+}
+
+Meteor.methods({
+
+    'workflow.setIsEditedMeetingSerie'(msId) {
+        console.log('workflow.setIsEditedMeetingSerie on ' + msId);
```

Comment:
@migerh What do you think: Is it necessary to have all those log-statements at the beginning and the end of each meteor method 🤔 ?

Reply:
Nevertheless, Code Climate marks this methods as duplicates correctly. So we should either get rid of the logs or get rid of the duplications. To get rid of the code duplicates we could define a function like this:
```
const logExecution(tag, action, ...args) {
    console.log(`START ${tag}`);
    action(...args);
    console.log(`DONE ${tag}`);
}
```
Then the body of the meteor methods would be just one single line. One example:
```
'workflow.removeIsEditedMeetingSerie'(msId, any) {
     logExecution(`workflow.removeIsEditedMeetingSerie on ${msID}`, removeIsEditedMS, msId, any);
 },
```

Reply:
> @migerh What do you think: Is it necessary to have all those log-statements at the beginning and the end of each meteor method  ?

No, we don't. Especially with this feature (locks will be set/deleted very, VERY often!) we spam the server logs a lot.